app/models/concerns/notifications/enum_status_values.rb

```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+
+# rubocop:disable Metrics/BlockLength, Metrics/MethodLength
+module Notifications
+  module EnumStatusValues
+    extend ActiveSupport::Concern
+
+    ACTIVEDUTY = 'activeduty'
```

Comment:
For sure.  Here was my rationale for defining the enum values as constants:

- it creates a single source of truth for the enum value definitions, that can be referenced elsewhere
- it enables a dev to reference any given enum value, from anywhere in the codebase, through the constant, vs. a standalone `:symbol`; this has the added benefit of confirming accuracy at runtime, with a very specific error should there be an issue
- enables us to be able to validate a given `subject` with its specific `statuses`; similar to a `has_many:belongs_to` db record association 
- It is more intentional about locking down the values themselves; IMO constants are generally accepted as "Hey developer, you should have good cause to change this value"
- somewhat mitigating against mistakenly mis-referencing elsewhere in the codebase

Here was my rationale for pulling the mappings into dedicated modules:

- by encapsulating all of the enum definitions, IMO it is more likely that future developers will read and follow the patterns that have been established, thereby mitigating potential errors/bugs
- I interpret them as a single responsibility, perfect for 'concern' encapsulation
- the list of values is quite long, and is intended to continue to keep growing
- creates a dedicated space for them to grow over time, without simultaneously cluttering up the model file with very large lists of constants; a byproduct of which would be continuing to push down the model callbacks/validations/scopes/methods further out of view

That work?

Reply:
> it creates a single source of truth for the enum value definitions, that can be referenced elsewhere

same applies to the enum only method, but with additional constants now you have two references for the same thing `Notification.statuses[:canceled_declined]` (1) and `Notification::CANCELED_DECLINED` ('canceled_declined'). With `Notification.statuses[:canceled_declined]` it's more clear that canceled_declined is a status option.

Reply:
> it enables a dev to reference any given enum value, from anywhere in the codebase, through the constant, vs. a standalone :symbol; this has the added benefit of confirming accuracy at runtime, with a very specific error should there be an issue

Referencing was covered above. It does create an error if you mistype a constant vs a hash key.

Reply:
> enables us to be able to create groupings to validate a given subject with its specific statuses; similar to a has_many:belongs_to db record association

the grouping `Notification.statuses` is created both ways.

Reply:
> It is more intentional about locking down the values themselves; IMO constants are generally accepted as "Hey developer, you should have good cause to change this value"

IMO enums have the same trait, but who knows 🤷‍♂️ 😀

Reply:
So yeah, part of the goal is to be able to access the actual enum string values elsewhere in the code.  

Using the constant you can get the values like this:

```
$ Notification::DASH_HCA
==> 'dashboard_health_care_application_notification'
```

Going through the `subject` or `status`, you can only get the database integer:

```
$ Notification::subjects[:dashboard_health_care_application_notification]
==> 1
```

Reply:
@kreek  If you're concerned with any of this, I can switch out the constants for symbols.  Shall I do so?

Reply:
On having dedicated modules, the primary concern of concerns is code reuse rather than to keep file length down. These values are unlikely to be reused elsewhere. And if there's not repeated constants the file will be shorter 😄

Reply:
Yes, that was one of the ancillary benefits.  The main drivers for that extraction that are:

- by encapsulating all of the enum definitions, IMO it is more likely that future developers will read and follow the patterns that have been established, thereby mitigating potential errors/bugs that can stem from `ActiveRecord::Enum` constraints
- I interpret them as a single responsibility, perfect for encapsulation

Instead of a concern, it could be a non-active-record backed model, a service class, or a lib class?

Or are you not seeing value at all there?

Reply:
My view is that you probably won't need the constant (string value) in isolation. So I would switch to symbols. Whether we go with constants symbolized as keys or use straights symbols Rails is like whatever and makes them string keys anyway (although with indifferent access so you can use symbols to look things up). And once you're setting the enum that way I don't see the need for the concern to be a data bag without any functionality.

Reply:
you can still make the statuses hash a constant like the operating hours for GIBS https://github.com/department-of-veterans-affairs/vets-api/blob/25bb8fade9195f336b8aa7d0a0265dea3f23093f/lib/evss/gi_bill_status/service.rb#L11-L15